x-pack/plugins/security_solution/public/common/components/hover_actions/use_hover_actions.tsx

```diff
@@ -50,8 +50,7 @@ export const useHoverActions = ({
   const [closePopOverTrigger, setClosePopOverTrigger] = useState(false);
   const [showTopN, setShowTopN] = useState<boolean>(false);
   const [hoverActionsOwnFocus, setHoverActionsOwnFocus] = useState<boolean>(false);
-  const [goGetTimelineId, setGoGetTimelineId] = useState(false);
-  const timelineIdFind = useGetTimelineId(containerRef, goGetTimelineId);
```

Comment:
we used to get timelineId from dom element with `data-timeline-id`, but this seem not able to get the right timelineId. Therefore instead of getting them from dom, I put context providers in those components which contains `data-timeline-id`, and use context here to get timelineId.

Reply:
Your solution using context looks way cleaner!